docker/studyu-cli

```diff
@@ -0,0 +1,516 @@
+  # local verbose="$3"
+
+  # If STUDYU_SECRETS_MANAGER is not empty, use it as the prefix for the Docker Compose command
+  local compose_command="docker compose -f $file $action"
+  if [ -n "$STUDYU_SECRETS_MANAGER" ]; then
+    compose_command="$STUDYU_SECRETS_MANAGER $compose_command"
+  fi
+
+  # Execute the Docker Compose command
+  # if [ "$verbose" == "--verbose" ]; then
+  $compose_command
+  # else
+  #  $compose_command >/dev/null 2>&1
+  # fi
+
+  # Check the exit status of the Docker Compose command and return an error if it failed
+  local exit_status=$?
+  if [ "$exit_status" -ne 0 ]; then
+    echo -e "${red}Error: Docker Compose command failed with exit status $exit_status.${reset}"
+    # echo -e "Run with --verbose for more information."
+    exit "$exit_status"
+  fi
+
+}
+
+
+# Run given docker compose commands such as start/restart/stop
+manage_components() {
+  local action=$1
+  local components="${*:2}" # Get components, excluding the first argument (action)
+
+  # If no components are specified, use the default components
+  if [ -z "${components[*]}" ]; then
+    components=("$STUDYU_DEFAULT_COMPONENTS")
+  fi
+
+  validate_component_list "${components[@]}"
+
+  for component in ${components[*]}; do
+    local compose_file="${compose_files[$component]}"
+    if [ -z "$compose_file" ]; then
+      echo -e "${red}Error: Invalid component '$component'${reset}"
+      echo -e "$available_components"
+      exit 1;
+    fi
+
+    if [[ "$action" == 'up'* && "$component" == 'supabase'* ]]; then
+      local supabase_path
+      supabase_path="$STUDYU_PATH/docker/supabase"
+      if [ ! -f "$supabase_path/.env" ]; then
+        echo -e "${yellow}Warning: Could not find a Supabase .env file at: $supabase_path${reset}"
+        read -rp "$(echo -e "${bold}Should a default .env file be created? (y/n): ${reset}")" create_supabase_env
+        if [[ "$create_supabase_env" =~ ^[Yy]$ ]]; then
+          cp "$supabase_path/.env.example" "$supabase_path/.env"
+          echo -e "${green}Supabase .env file successfully created at: $supabase_path/.env${reset}"
+          echo -e "Please modify the default secrets before starting the containers for the first time."
+          exit 0
+        else
+          echo -e "${red}Error: Supabase .env file is needed to run.${reset}"
+          exit 1
+        fi
+      fi
+    fi
+    run_docker_compose "$compose_file" "$action"
+  done
+}
+
+# Check if components are running healthy
+status() {
+  container_names=$(docker ps --filter "name=supabase" --filter "name=studyu" --format "{{.Names}}")
+  for name in $container_names; do
+      health_status=$(docker inspect --format='{{.State.Running}}' "$name")
+      if [[ "$health_status" != "true" ]]; then
+          echo -e "${red}Error: $name is not running healthy. Check 'docker compose ps' for troubleshooting.${reset}"
+          exit 1;
+      fi
+  done
+  echo -e "${green}All components are running healthy.${reset}"
+  exit 0;
+}
```

Comment:
This says everything is running healthy when no containers are running; maybe we can first verify that everything that is configured to run is running at all